src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -103,6 +188,30 @@ export class SamDebugConfigProvider implements vscode.DebugConfigurationProvider
         folder: vscode.WorkspaceFolder | undefined,
         config: AwsSamDebuggerConfiguration,
         token?: vscode.CancellationToken
+    ): Promise<SamLaunchRequestArgs | undefined> {
+        const resolvedConfig = await this.makeConfig(folder, config, token)
+        if (!resolvedConfig) {
+            return undefined
+        }
+        await this.invokeConfig(resolvedConfig)
```

Comment:
`resolveDebugConfiguration` is implemented by two functions: `makeConfig` and `invokeConfig`. This allows `makeConfig` to be tested without invoking sam/docker/debugger.